src/main/java/wooteco/subway/line/api/LineController.java

```diff
@@ -0,0 +1,51 @@
+package wooteco.subway.line.api;
+
+import javax.validation.Valid;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.net.URI;
+import java.util.List;
+import wooteco.subway.line.api.dto.LineRequest;
+import wooteco.subway.line.api.dto.LineResponse;
+import wooteco.subway.line.service.LineService;
+
+@RestController
+public class LineController {
+
+    private final LineService lineService;
+
+    public LineController(LineService lineService) {
+        this.lineService = lineService;
+    }
+
+    @PostMapping("/lines")
+    public ResponseEntity<LineResponse> createLine(@RequestBody @Valid LineRequest lineRequest) {
+        LineResponse newLine = lineService.createLine(lineRequest);
+        return ResponseEntity.created(URI.create("/lines/" + newLine.getId())).body(newLine);
```

Comment:
```suggestion
        return ResponseEntity.created(URI.create("/lines/" + newLine.getId()))
        .body(newLine);
```

메서드 체이닝은 내려쓰는것이 가독성과 디버깅 측면에서 모두 좋은 것 같아요.
26 라인에서 에러가 발생한다면 어떤 행위에서 에러가 발생하는지 확인하기 어려울 것 같아요. 🤔

그런 의미에서 URI.create("/lines/" + lineResponse.getId()) 도 변수로 분리되면 장/단점이 있겠네요.

Reply:
그 동안은 inline 방식 vs 변수 추출에 대해서 단지 메서드 길이, 혹은 가독성에 대한 이슈로만 생각하고 있었는데,
디버깅이라는 관점으로도 볼 수 있겠네요! 
새로운 인사이트 주셔서 감사합니다.

그런데 그 밑에 `return ResponseEntity.ok().body(dto);` 부분만 본다면 `ResponseEntity.ok()` 까지는 에러가 발생할 일이 거의 없을 것 같고, 에러가 발생한다면 그 뒤의 body를 넣어주는 부분을 의심해볼 수 있을 것 같은데 (사실 body() 메서드도 뜯어보니 @Nullable 이라 에러가 발생할 경우가 현재는 잘 생각나지 않긴 하지만요...), 
이런 경우에도 미립은 (가독성 혹은 다른 이유에 의해) 내려쓰시는 편인가요?

Reply:
저는 [규칙 4: 한 줄에 한 점만 사용](https://developerfarm.wordpress.com/2012/01/30/object_calisthenics_5/) 을 지키려고 노력하는 편이에요 말씀하신것처럼 에러가 발생하지 않을 수 있더라도 여러 행위를 하는 것을 가독성 있게 볼 수 있는 장점도 있다고 생각해요.

Reply:
추가로 행위 단위로 내려쓴다는 `일관성`도 유지하려 하고 있구요.